### DISCUSSION TITLE
HDFS support in rust

### DISCUSSION THREAD

**Discussion by nddashti** [delta-io/delta-rs#2954] [Q&A] [answered]
I'm trying read deltatable from hdfs but I received error:

"Cannot infer storage location from: Unknown scheme: hdfs"

My code is:

let endpoint_url = env.hdfs_endpoint_url.ok_or_else(|| not_set_env("HDFS Endpoint Url"))?;
let directory_path = env.hdfs_directory_path.ok_or_else(|| not_set_env("Directory Path"))?;
let data_url = url::Url::parse(&format!("{}/{}/{}", endpoint_url, directory_path, df_name)).unwrap();
let data_table: deltalake::DeltaTable = open_table(data_url).await;

**Comment by rtyler** [comment, marked as answer]
@nddashti the HDFS handler has not been registered, e.g. 

```rust
deltalake_hdfs::register_handlers(None);
```
